src/libudev/libudev.c

```diff
@@ -109,16 +103,9 @@ _public_ struct udev *udev_ref(struct udev *udev) {
  * Drop a reference of the udev library context. If the refcount
  * reaches zero, the resources of the context will be released.
  *
- * Returns: the passed udev library context if it has still an active reference, or #NULL otherwise.
+ * Returns: #NULL.
  **/
-_public_ struct udev *udev_unref(struct udev *udev) {
-        if (udev == NULL)
-                return NULL;
-        udev->refcount--;
-        if (udev->refcount > 0)
-                return udev;
-        return mfree(udev);
```

Comment:
Nope, this is not backwards compatible.

(For example, someone could do cleanup in this way:
```C
while ((udev = udev_unref(udev)));
```
With this change, this becomes broken.)

Reply:
OK. Note that the man page udev_unref(3) states that 'udev_unref() always returns NULL.' Should we update this?